### ISSUE TITLE
Add ability to automatically download roles when running play

### ISSUE THREAD

**Issue by bcoca** [ansible/ansible#12927]
##### Issue Type:

Feature Idea
##### Ansible Version:

Future
##### Ansible Configuration:

galaxy_whitelist: galaxy.ansible.com, myrepo.private.com
##### Summary:

Allow ansible-playbook auto-download roles when executing play if the role source is white-listed and the auto-download feature is enabled (off by default).

**Comment by smiller171** [comment]
:+1:

**Comment by msabramo** [comment]
:+1: 

Interestingly, https://galaxy.ansible.com/intro has some verbiage that makes it sound like Ansible already automatically downloads role dependencies even though this is not implemented yet. 

> Ansible uses the dependencies section of meta/main.yml to download and apply any dependent roles.
> Add one or more dependent roles that Ansible will download and execute before running your role. 

The word _download_ to me seems to suggest that Ansible already does what this issue purports to do, but it doesn't work, though I will admit that when I stumbled upon this in the docs, I got excited because I assumed it had been implemented in Ansible 2.0 and I hadn't noticed. That excitement quickly turned to disappointment when I tried it out with 2.0 and I got the same old error about not being able to find the role. 

I would suggest revising the docs to be clearer about this. On second thought, it would just be better to implement this to make the docs true :-)
